.github/workflows/_local-set-default-track.yaml

```diff
@@ -0,0 +1,40 @@
+name: Set charm default track
+
+on:
+  workflow_call:
+    inputs:
+      charm-name:
+        description: "Path to the charm we want to promote. Defaults to the current working directory."
+        type: string
+        required: true
+      target-track:
+        description: "Track name to set as 'default track'."
+        type: string
+        required: true
+    secrets:
+      CHARMHUB_TOKEN:
+        required: true
+
+jobs:
+  set-track:
+    name: Set default track
+    runs-on: ubuntu-24.04
+    env:
+      CHARMHUB_TOKEN: ${{ secrets.CHARMHUB_TOKEN }}
+      CHARM_NAME: ${{ inputs.charm-name }}
+      TARGET_TRACK: ${{ inputs.target-track }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install dependencies
+        run: |
+          sudo snap install http
```

Comment:
Snap seems abandoned and not up do date with [upstream](https://github.com/httpie/cli).
Should we use curl? [Example](https://discourse.charmhub.io/t/obtain-all-associated-revisions-via-stores-api/13527).